drivers/i2c/i2c_ll_stm32.h

```diff
@@ -11,13 +11,31 @@
 
 typedef void (*irq_config_func_t)(struct device *port);
 
+#ifdef CONFIG_I2C_STM32_V2
+/**
+ * @brief structure to convey optional i2c timings settings
+ */
+struct i2c_config_timing {
+	/* i2c periphral clock in Hz */
+	uint32_t periph_clock;
+	/* i2c bus speed in Hz */
+	uint32_t i2c_speed;
+	/* I2C_TIMINGR register value of i2c v2 peripheral */
+	uint32_t timing_setting;
+};
+#endif
+
 struct i2c_stm32_config {
 #ifdef CONFIG_I2C_STM32_INTERRUPT
 	irq_config_func_t irq_config_func;
 #endif
 	struct stm32_pclken pclken;
 	I2C_TypeDef *i2c;
 	uint32_t bitrate;
+#ifdef CONFIG_I2C_STM32_V2
+	const struct i2c_config_timing *timings_list;
```

Comment:
`_list` prefix is redundant as `timings` is plural, implying > 1.

Reply:
and that will make lines even shorter - done